src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -291,6 +291,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let offset_ptr = ptr.ptr_wrapping_signed_offset(offset_bytes, self);
                 self.write_scalar(offset_ptr, dest)?;
             }
+            sym::ptr_guaranteed_eq | sym::ptr_guaranteed_ne => {
+                // FIXME: return `true` for at least some comparisons where we know equality and
+                // inequality will always be equal to the runtime equivalent.
```

Comment:
"equality equal to the equivalent"? Lots of equals. ;)
```suggestion
                // FIXME: return `true` for at least some comparisons where we can reliably
                // determine the result of runtime (in)equality tests at compile-time.
```

Reply:
I think I suggested something like this before, and I still think "reliably determine runtime result" is the clearest way to express this, which is why I suggested it both here and for the doc comment. Not sure why you did not like it here?

Reply:
Sorry, your review was good, I just didn't read it properly :/